readme.md

```diff
@@ -168,6 +168,21 @@ document.body.append(
 );
 ```
 
+To improve compatibility with React components, `dom-chef` will pass the function's `defaultProps` property to itself (if present). Note that specifying attributes won't override those defaults, but instead set them on the resulting element:
+
+```jsx
+function AlertIcon(props) {
+	return <svg width={props.size} className={props.className} />
+}
+
+AlertIcon.defaultProps = {
+	className: 'icon icon-alert'
+	size: 16,
+}
+
+const el = <AlertIcon className="margin-0" size={32} />;
+// <svg width="16" class="icon icon-alert margin-0" size="32" />
```

Comment:
Hm, how does `class` end up with `icon icon-alert margin-0`, when `<AlertIcon>` passed only `margin-0`? How did default value get joined there?

Reply:
Line 178 of this file

Reply:
Line 178 of this file

Reply:
Yeah, there's a default value for `className` prop, but how does it get merged with `className` that's passed to the component? `<Alert className="margin-0"> => <svg className="icon icon-alert margin-0">`. In React it would overwrite default prop, but in this example it doesn't seem to somehow.

Reply:
Indeed, I didn't realize that `defaultProps` would be overridden in React.

I guess that the "correct" usage would be:

```jsx
 const el = <AlertIcon className={AlertIcon.defaultProps.className + ' margin-0'} size={32} />;
```

hmmm...

Reply:
The merging happens with all elements:

https://github.com/vadimdemedes/dom-chef/blob/1365a05fa7ce343082cb34caf9dd2717ac32b728/index.ts#L94-L95

Reply:
Yes but we should match React’s behavior when possible. DefaultProps are overridden one React, not merged

Reply:
We can't override the props at all; we can just work with the DOM element that is created by the functional component (see [`bb8f576`, line 171](https://github.com/vadimdemedes/dom-chef/pull/63/commits/bb8f576c3de00f4c662bfb9060282cc94a3a5590#diff-0730bb7c2e8f9ea2438b52e419dd86c9R171)):

> Note that specifying attributes won't override those defaults, but instead set them on the resulting element.

Then it should behave exactly the same as other functions that return a DOM element.

Reply:
Yeah, the way I set it up in my "function components" PR was specifically to merge classes because _we_ (Refined GitHub devs) find it useful: https://github.com/vadimdemedes/dom-chef/pull/59

Also in that PR I suggested possible future attributes to change this behavior, but until anyone needs that, we can just implement it in the most comfortable way (to us).

So, LGTM